src/cmap/connection.ts

```diff
@@ -293,37 +292,57 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     this[kLastUseTime] = now();
   }
 
-  onError(error: Error) {
+  _cleanup(options: { force: boolean; errorFactory?: () => Error }) {
```

Comment:
errorFactory could be changed, this maintains the old behavior each callback receives a different instance of the error. We could also just passed an error indirectly.

Reply:
Yeah, it’s not immediately obvious to me why this isn’t `error: Error` instead of `errorFactory`… is it for the stack trace? That would make sense but is probably something that should be noted in a comment

Reply:
This technically preserves the existing behavior where each callback receives a distinct instance of the error. I think it's a reasonable argument to change this behavior (and good point about to call stack) so if we're OK passing the same instance to each call back, then i'll make that change.

_Technically_ there should only be one call back in there ever though.

Reply:
Right, if object identity for the `Error` objects is the concern here, I’m okay with passing the same `Error` object to multiple callbacks.